### ISSUE TITLE
RecurrentLatentWorldModelEnsemble: max_raw_gradient_norm rejects legal in-bounds transitions after ordinary training and the rejection is a permanent deadlock

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#366]
## Summary (contract question — no PR attached; the fix is a design decision)

`RecurrentLatentWorldModelEnsembleConfig` (`alberta_framework/core/recurrent_latent_world_model_ensemble.py`) documents the `max_*` fields as "fail-closed validity gates, not tuned hyperparameters … coarse sanity ceilings far above any healthy operating range". `max_raw_gradient_norm = 100_000` is compared against each member's raw NLL gradient (`member_gradients_valid`), and a single member breaching it vetoes the whole event, leaving persistent state bit-identical.

The reachable raw NLL gradient scales as `residual / variance`, and `variance_floor = 1e-3` puts a 1000× multiplier on it. Nothing at construction relates `max_raw_gradient_norm` to `max_input_magnitude` (1000) or `variance_floor`. So once the heteroscedastic head has correctly learned that a stream is low-noise, a transition whose every value is far inside `max_input_magnitude` blows through the ceiling — and because rejection leaves state unchanged, the identical event is rejected again forever. `gradient_clip_norm = 10` already exists to bound the committed step (and `candidate_parameters_valid` independently confirms it is benign), so the raw-gradient ceiling rejects exactly the case the clipper was built for.

Two consequences for measurement in a continual-learning module where the distribution shift *is* the measurement:

1. The model deadlocks at the shift, silently, while every input-validity, ownership, boundary, and capacity diagnostic reports healthy.
2. `_rejected_result` returns `next_start_cache = _zero_start_cache()`; the module's own `test_jit_and_scan_match_sequential_cache_state_and_outputs` carries `result.next_start_cache` in the scan carry, so after one rejected event every later `decide` returns a zero cache and every later `update` is rejected — the run is frozen from that step on. Each rejected step also emits `mean_negative_log_likelihood = 0.0` (no availability flag other than `diagnostics.applied`), so a windowed NLL over a stalled run reads *better* than the healthy stationary window (0.0 vs −0.976 in the repro).

## Reproduction (deterministic, pristine `main` @ `45279f3`; all bounds and rates at their defaults)

`observation_dim=2, n_actions=2, latent_dim=8, ensemble_size=3, uncertainty_warmup_steps=0, max_updates=100000`, everything else default (`learning_rate=1e-3`, `bootstrap_probability=0.8`, `max_raw_gradient_norm=1e5`, `max_input_magnitude=1000`). Train 30k events on a fixed low-noise transition (`obs=(0.25,-0.5) → bootstrap=(0.75,0.125)`, reward 0.5, discount 0.9), then present a legal shifted bootstrap observation:

```
after 30000 default-rate updates, learned variances: [0.00113 0.00114 0.00113 0.00116]
  legal shift +-10     (ceiling 1000) applied=True  grads_valid=[True, True, True]
  legal shift +-50     (ceiling 1000) applied=False grads_valid=[False, False, False]
  legal shift +-100    (ceiling 1000) applied=False grads_valid=[False, False, False]
  legal shift +-200    (ceiling 1000) applied=False grads_valid=[False, False, False]
```

The threshold sits near residual ≈ 42, i.e. ~4% of the legal input range (member grad norms `[36134, 143379, 69644]` at ±50 vs ceiling 100000). Repeating the same legal ±50 transition: `50/50 silently rejected; event_count frozen | every other gate: state True input True ownership True boundary True capacity True`. Carrying `next_start_cache` like the scan loop after that single rejection: `events applied after the rejected shift: 0/29`.

## Options (for the maintainer to choose)

- (a) make `member_gradients_valid` a finiteness gate only and let `gradient_clip_norm` + the existing `candidate_parameters_valid` / `candidate_state_valid` gates bound the committed step; or
- (b) validate at construction that `max_raw_gradient_norm` covers the worst legal raw gradient implied by `max_input_magnitude` and `variance_floor` (which would reject the current defaults); and, either way,
- (c) on a rejection where `state_valid & cache_valid` hold, return a `next_start_cache` that re-owns the unchanged state so a transient rejection is retryable rather than terminal, and give the loss channel an explicit availability flag (mirroring `representation_gradient_available`).

I have not opened a PR because (a) vs (b) changes what the gate means; happy to implement whichever you pick, failing-test-first.

## Dedup

Related open (mine): #333/#336 (init state validity in the same module — different mechanism). `gh issue/pr list --search` for "max_raw_gradient_norm", "gradient norm reject", "next_start_cache", "world model ensemble": only #139/#140 (checkpoint format, unrelated) and PR #61 (ObGD 0×inf).
